### PR TITLE
serial_terminal: die in case unsuccessful login

### PR DESCRIPTION
**Pull request by asmorodskyi**
die in case unsuccessful login

VR: http://10.160.67.22/tests/47#step/before_test/4

### CODE REVIEW

**Comment by asmorodskyi** [comment]
Verification run added , @richiejp , @pevik WDYT ?

File: lib/serial_terminal.pm [hunk 1]
```diff
@@ -84,12 +84,12 @@ sub login {
...
     type_string("\n");
-    wait_serial(qr/login:\s*$/i);
+    die 'Failed to wait for login prompt' unless wait_serial(qr/login:\s*$/i);
     type_string("$user\n");
-    wait_serial(qr/Password:\s*$/i);
+    die 'Failed to wait for password prompt' unless wait_serial(qr/Password:\s*$/i);
     type_password;
     type_string("\n");
-    wait_serial(qr/$escseq* \w+:~\s\# $escseq* \s*$/x);
+    die 'Failed to confirm that login successful' unless wait_serial(qr/$escseq* \w+:~\s\# $escseq* \s*$/x);
```
Comment (pevik) [inline lib/serial_terminal.pm]: nit: 'Failed to confirm that login *was* successful'

**Comment by richiejp** [comment]
LGTM, except for Pter's nit.

File: lib/serial_terminal.pm (hunk 1 above)
Comment (asmorodskyi) [reply lib/serial_terminal.pm]: Done

**Comment by DimStar77** [comment]
This one seems to be relevant for the new failures on the openSUSE Live images:

https://openqa.opensuse.org/tests/1057088#step/sshd/20

root and user have no password set there, so 'root' can login without a password prompt - yet, the test now dies out.

**Comment by okurz** [comment]
@asmorodskyi please be aware of the revert in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8681 to fix the current tests and please look into the issue together with bringing back the content of the PR because certainly it is very useful to not ignore the failed checks.

**Pull request merged**